source/_docs/automation/trigger.markdown

```diff
@@ -904,6 +903,12 @@ Note that to use JSON encoded payloads, the `Content-Type` header must be set to
 ```bash
 curl -X POST -H "Content-Type: application/json" -d '{ "key": "value" }' https://your-home-assistant:8123/api/webhook/some_hook_id
 ```
+If you want to pass webhook data to action, you will need to use something like this in your action
+```yaml
+data_template:
+  payload: "{{trigger.json}}"
+```
+Note: if you pass json when template is expecting non-json data or vice versa, then automation will trigger but payload will be empty.
```

Comment:
That sounds like a bug, please report it, as Home Assistant shouldn't just lose data or raise an error instead.